tensorflow/java/src/test/java/org/tensorflow/TensorTest.java

```diff
@@ -105,55 +116,90 @@ public void createFromBufferWithNonNativeByteOrder() {
-        long[] actual = new long[longs.length];
-        assertArrayEquals(longs, t.copyTo(actual));
-      }
+  @Test
+  public void whenCreateUsingFloatBufferThenExpectedMatrixIsStoredInTensor() {
+    final float[] expected = {1f, 2f, 3f, 4f};
+    final long[] shape = {4};
+
+    try (Tensor t = Tensor.create(shape, FloatBuffer.wrap(expected))) {
+      final float[] actual = new float[expected.length];
+      t.copyTo(actual);
+      assertArrayEquals(expected, actual, EPSILON_F);
+    }
+  }
+  
+  @Test
+  public void whenCreateUsingIntBufferThenExpectedMatrixIsStoredInTensor() {
+    final int[] expected = {1, 2, 3, 4};
+    final long[] shape = {4};
+
+    try (Tensor t = Tensor.create(shape, IntBuffer.wrap(expected))) {
+      final int[] actual = new int[expected.length];
+      t.copyTo(actual);
+      assertArrayEquals(expected, actual);
     }
+  }
 
-    // validate shape-checking
-    {
-      try (Tensor t = Tensor.create(new long[doubles.length + 1], DoubleBuffer.wrap(doubles))) {
-        fail("should have failed on incompatible buffer");
-      } catch (IllegalArgumentException e) {
-        // expected
-      }
-      try (Tensor t = Tensor.create(new long[floats.length + 1], FloatBuffer.wrap(floats))) {
-        fail("should have failed on incompatible buffer");
-      } catch (IllegalArgumentException e) {
-        // expected
-      }
-      try (Tensor t = Tensor.create(new long[ints.length + 1], IntBuffer.wrap(ints))) {
-        fail("should have failed on incompatible buffer");
-      } catch (IllegalArgumentException e) {
-        // expected
-      }
-      try (Tensor t = Tensor.create(new long[longs.length + 1], LongBuffer.wrap(longs))) {
-        fail("should have failed on incompatible buffer");
-      } catch (IllegalArgumentException e) {
-        // expected
-      }
+  @Test
+  public void whenCreateUsingLongBufferThenExpectedMatrixIsStoredInTensor() {
+    final long[] expected = {1L, 2L, 3L, 4L};
+    final long[] shape = {4};
+
+    try (Tensor t = Tensor.create(shape, LongBuffer.wrap(expected))) {
+      final long[] actual = new long[expected.length];
+      t.copyTo(actual);
+      assertArrayEquals(expected, actual);
+    }
+  }
+
+  @Test
+  public void whenCreateUsingDoubleBufferWithIncorrectShapeThenThrowExpectedException() {
+    expectedException.expect(IllegalArgumentException.class);
+    expectedException.expectMessage("buffer with 4 elements is not compatible with a Tensor with shape [0, 0, 0, 0, 0]");
+    final double[] expected = {1D, 2D, 3D, 4D};
+
+    try(Tensor t = Tensor.create(new long[expected.length + 1], DoubleBuffer.wrap(expected))) {
+    }
+  }
+
+  @Test
+  public void whenCreateUsingFloatBufferWithIncorrectShapeThenThrowExpectedException() {
+    expectedException.expect(IllegalArgumentException.class);
+    expectedException.expectMessage("buffer with 4 elements is not compatible with a Tensor with shape [0, 0, 0, 0, 0]");
```

Comment:
Now that the tests are broken up and we're not re-using the same testdata, perhaps these can be simplified (e.g., using: `expected = {1F};` instead of a longer array)